tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -3317,16 +3317,18 @@ mlir::Value MLIRScanner::GetAddressOfDerivedClass(
     }
 
     mlir::Value ptr = value;
-    if (auto PT = ptr.getType().dyn_cast<LLVM::LLVMPointerType>())
+    if (auto PT = ptr.getType().dyn_cast<LLVM::LLVMPointerType>()) {
       ptr = builder.create<LLVM::BitcastOp>(
           loc,
           LLVM::LLVMPointerType::get(builder.getI8Type(), PT.getAddressSpace()),
           ptr);
-    else
+    } else {
+      unsigned memorySpace =
+          ptr.getType().cast<mlir::MemRefType>().getMemorySpaceAsInt();
       ptr = builder.create<polygeist::Memref2PointerOp>(
-          loc,
-          LLVM::LLVMPointerType::get(builder.getI8Type(), PT.getAddressSpace()),
+          loc, LLVM::LLVMPointerType::get(builder.getI8Type(), memorySpace),
```

Comment:
I assume the pointer is either an llvm pointer or a memref.

Reply:
Yes